src/Angor/Shared/Services/SignService.cs

```diff
@@ -82,6 +83,7 @@ public Task LookupInvestmentRequestsAsync(string nostrPubKey, string? senderNpub
             {
                 var subscription = nostrClient.Streams.EventStream
                     .Where(_ => _.Subscription == subscriptionKey)
+                    .Where(_ => _.Event.Tags.FindFirstTagValue("subject") == "Investment offer")
```

Comment:
@DavidGershony what do you think about this as filter on the subscription?
I suppose the filter only actually happens in the client not the server, I am not sure it is possible to filter based on subject.

Reply:
It happens on the client side, if you want to filter on the server side you need to add it to the filter we send to the relay, and you can add tags to the filter so should be possible to filter by subject.

Reply:
I will try this next